dev/generate-kernel-signatures.py

```diff
@@ -4,12 +4,21 @@
 
 import os
 import datetime
+import time
 
 import yaml
 
 CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
 
 
+def reproducible_datetime():
+
+    build_date = datetime.datetime.utcfromtimestamp(
+        int(os.environ.get("SOURCE_DATE_EPOCH", time.time()))
+    )
+    return build_date.isoformat().replace("T", " AT ")[:22]
```

Comment:
Small question (probably @jpivarski), can we just specify [`timespec`](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat) instead of truncating to 22 characters? If this time is not parsed anywhere, then `timespec="milliseconds"` would probably suffice?

Reply:
`timespec` was introduced in Python 3.6, and (at the moment) we're still supporting Python 2.7. The ISO format ensures that cutting the string at a specific number of characters is always correct.

Reply:
I *always* forget Py2, that's not good!

Reply:
For Awkward Array, it won't matter next year: #1010.